chore: update doc (#914)
instill-ai/instill.tech: docs/component/application/email.en.mdx

---
title: "Email"
lang: "en-US"
draft: false
description: "Learn about how to set up a VDP Email component https://github.com/instill-ai/instill-core"
---

> The Email component is a application component that allows users to get and send email from Mail Protocol.

Reviewer: ```suggestion
The Email component is an application component that allows users to get and send email from Mail Protocol.
```

It can carry out the following tasks:

- [Send Email](#send-email)
- [Read Emails](#read-emails)

> You can connect different email server through Email component.

Reviewer: ```suggestion
You can connect to different email servers through the Email component.
```

Emails are fetched and sent using the IMAP and SMTP protocols, respectively. You can set the server address and port for each protocol in the component configuration.

## Release Stage

`Alpha`

## Configuration

The component configuration is defined and maintained [here](https://github.com/instill-ai/component/blob/main/application/email/v0/config/definition.json).

## Setup

| Field | Field ID | Type | Note |
| :--- | :--- | :--- | :--- |
| Server Address (required) | `server-address` | string | The address of the email server |
| Server Port (required) | `server-port` | integer | The port of the email server |

> | Email Address (required) | `email-address` | string | Email address of the user |
> | App Password (required) | `password` | string | Password of the App passwords in Gmail settings |

Reviewer: ```suggestion
| Email Address (required) | `email-address` | string | The email address of the user |
| App Password (required) | `password` | string | The password of the App passwords in Gmail settings |
```

 - 2. Create a new App Password and save it in a secure place.
 - 3. Add App Password as a new secret in the Instill Platform by navigating to **Console** > **Settings** > **Secrets**.
 - 4. Reference the secret in the App Password field in the component configuration.

## Supported Tasks

### Send Email

Send an email to recipients

| Input | ID | Type | Description |
| :--- | :--- | :--- | :--- |
| Task ID (required) | `task` | string | `TASK_SEND_EMAIL` |
| Recipient (required) | `recipients` | array[string] | The email addresses of the recipients |
| Cc | `cc` | array[string] | The email addresses for Carbon Copy |
| Bcc | `bcc` | array[string] | The email addresses for Blind Carbon Copy |
| Subject | `subject` | string | The subject of the email |
| Message (required) | `message` | string | The message to be sent |

| Output | ID | Type | Description |
| :--- | :--- | :--- | :--- |
| Result | `result` | string | The result of sending the email |

### Read Emails

Read emails from a mailbox

| Input | ID | Type | Description |
| :--- | :--- | :--- | :--- |
| Task ID (required) | `task` | string | `TASK_READ_EMAILS` |

> | Search | `search` | object | the search criteria for the emails |

Reviewer: ```suggestion
| Search | `search` | object | The search criteria for the emails |
```
